File: archive/count-the-number-of-consistent-strings.js
const countConsistentStrings = function(allowed, words) {
  allowed = allowed.split('')

  let counter = 0
  let ans = words.map(word => {
    let memWord = word
    allowed.forEach(fil => {
      memWord = memWord.split('').filter((letter) => {
        if (letter === fil) {
          return ''
        } else {
          return letter
        }
      }).join('')
    })

    if (memWord.length === 0) {
      counter++
    }
    return memWord
  })

  return counter
};

console.log(countConsistentStrings( "abc", ["a","b","c","ab","ac","bc","abc"]))

File: go-archive/count-the-number-of-consistent-strings.go
func countConsistentStrings(allowed string, words []string) int {
    ans:=0

    mem:=map[byte]bool{}

    for i:=0; i <len(allowed); i++ {
        mem[allowed[i]] = true
    }

    for i:=0; i < len(words); i++ {
        m:=map[byte]bool{}

        for j:=0; j < len(words[i]); j++ {
            m[words[i][j]] = true
        }

        isOk:=true
        for k:= range m {
            if (!mem[k]) {isOk = false}
        }

        if isOk {ans++}
    }

    return ans
}